internal/impl/qdrant/output.go

```diff
@@ -103,36 +103,36 @@ type outputWriter struct {
 }
 
 func newOutputWriter(conf *service.ParsedConfig, mgr *service.Resources) (*outputWriter, error) {
-	collectionName, err := conf.FieldInterpolatedString(poFieldCollectionName)
+	collectionName, err := conf.FieldInterpolatedString(qoFieldCollectionName)
 	if err != nil {
 		return nil, err
 	}
 
-	host, err := conf.FieldString(poFieldGrpcHost)
+	host, err := conf.FieldString(qoFieldGrpcHost)
 	if err != nil {
 		return nil, err
 	}
 
-	apiToken, err := conf.FieldString(poFieldAPIToken)
+	apiToken, err := conf.FieldString(qoFieldAPIToken)
 	if err != nil {
 		return nil, err
 	}
 
-	config, enabled, err := conf.FieldTLSToggled(poFieldUseTLS)
+	config, enabled, err := conf.FieldTLSToggled(qoFieldUseTLS)
 	if err != nil {
 		return nil, err
 	}
-	id, err := conf.FieldBloblang(poFieldID)
+	id, err := conf.FieldBloblang(qoFieldID)
 	if err != nil {
 		return nil, err
 	}
 
-	vectorMapping, err := conf.FieldBloblang(poFieldVectorMapping)
+	vectorMapping, err := conf.FieldBloblang(qoFieldVectorMapping)
 	if err != nil {
 		return nil, err
 	}
 
-	payloadMapping, err := conf.FieldBloblang(poFieldPayloadMapping)
+	payloadMapping, err := conf.FieldBloblang(qoFieldPayloadMapping)
 	if err != nil {
 		return nil, err
 	}
```

Comment:
If we go back to making this optional then you need:

```
var payloadMapping service.BloblangExecutor*
if conf.Contains(qoFieldPayloadMapping) {
  payloadMapping, err = conf.FieldBloblang(qoFieldPayloadMapping)
  if err != nil {
    return nil, err
  }
}
```

Then you just need to handle this being `nil` while writing outputs and omit the payload (or write nil or something). You can also default to `root = {}` so you write an empty object by default.

Reply:
Default would be simpler, yeah.
Is that
```
.Default(`root = {}`)
```
for a `BloblangField`?

Reply:
[`1ff5a1c` (#2772)](https://github.com/redpanda-data/connect/pull/2772/commits/1ff5a1c0c889a8e0c8a2a803c571b342381ad1b7)

Reply:
Yes the default is right